lib/src/realm_class.dart

```diff
@@ -110,12 +112,58 @@ class Realm implements Finalizable {
   /// Opens a `Realm` using a [Configuration] object.
   Realm(Configuration config) : this._(config);
 
-  Realm._(this.config, [RealmHandle? handle, this._isInMigration = false]) : _handle = handle ?? _openRealm(config) {
+  Realm._(this.config, [RealmHandle? handle, this._isInMigration = false]) : _handle = handle ?? _openRealmSync(config) {
     _populateMetadata();
     isFrozen = realmCore.isFrozen(this);
   }
 
-  static RealmHandle _openRealm(Configuration config) {
+  /// A method for asynchronously opening a [Realm].
+  ///
+  /// If the configuration is [FlexibleSyncConfiguration], the realm will be downloaded and fully
+  /// synchronized with the server prior to the completion of the returned [Future].
+  /// This method could be called also for opening a local [Realm].
+  ///
+  /// * `config`- a configuration object that describes the realm.
+  /// * `cancellationToken` - an optional [CancellationToken] used to cancel the operation.
+  /// * `onProgressCallback` - a callback for receiving download progress notifications for synced [Realm]s.
+  ///
+  /// Returns [Future<Realm>] that completes with the `realm` once the remote realm is fully synchronized or with a [CancelledException] if operation is canceled.
+  /// When the configuration is [LocalConfiguration] this completes right after the local realm is opened or operation is canceled.
+  static Future<Realm> open(Configuration config, {CancellationToken? cancellationToken, ProgressCallback? onProgressCallback}) async {
+    Realm realm = (() => Realm(config)).asCancellable(
+      cancellationToken,
+      onCancel: (result) => result?.close(),
+    );
```

Comment:
I created `asCancellable` not only for futures (which come from the `cancellation_token` package), but also for synchronous functions. That way, whenever the token is canceled, we can release the resource that was returned by the function. Even if `cancel` is called before or after the main function.
```Dart
    Realm realm = (() => Realm(config)).asCancellable(
      cancellationToken,
      onCancel: (result) => result?.close(),
    );
```